tests/test_globalization.py

```diff
@@ -3,91 +3,148 @@
-                        .format(table_name, col1_name, col2_name, test_str)
-                    
-                    for rows, columns, status, statement, is_error in client.execute_query(setup_query):
-                        assert is_error == False
-
-                    select_query = u"SELECT {0}, {1} FROM {2};".format(col1_name, col2_name, table_name)
-                    for rows, columns, status, statement, is_error in client.execute_query(select_query):
-                        assert is_error == False
-                        assert len(rows) == 2
-                        assert decode(rows[0][0]) == u'value_{0}1'.format(test_str)
-                        assert decode(rows[1][0]) == u'value_{0}2'.format(test_str)
+            for characters in characters_list:
+                test_str = characters
+                col1_name = u'col_{0}1'.format(test_str)
+                col2_name = u'col_{0}2'.format(test_str)
+                table_name = u'#mssqlcli_{0}_{1}_{2}'.format(local_machine_name, random_str(), test_str)
+                setup_query = u"CREATE TABLE {0} ({1} nvarchar(MAX), {2} int);"\
+                    u"INSERT INTO {0} VALUES (N'value_{3}1', 1);"\
+                    u"INSERT INTO {0} VALUES (N'value_{3}2', 2);"\
+                    .format(table_name, col1_name, col2_name, test_str)
+                
+                for rows, columns, status, statement, is_error in client.execute_query(setup_query):
+                    assert is_error == False
+
+                select_query = u"SELECT {0}, {1} FROM {2};".format(col1_name, col2_name, table_name)
+                for rows, columns, status, statement, is_error in client.execute_query(select_query):
+                    assert is_error == False
+                    assert len(rows) == 2
+                    assert decode(rows[0][0]) == u'value_{0}1'.format(test_str)
+                    assert decode(rows[1][0]) == u'value_{0}2'.format(test_str)
         finally:
             shutdown(client)
 
 
-    def get_charset_double(self):
-        return [
+class GlobalizationMetadataTests(unittest.TestCase):
+
+    def test_schema_metadata_double(self):
+        charset = GB18030()
+        characters_list = charset.\
+            get_next_characters(charset.get_double_characters())
+        self.run_schema_metadata_validation(characters_list)
+
+
+    def run_schema_metadata_validation(self, characters_list):
+        client = create_mssql_cli_client()
+        completer = MssqlCompleter(smart_completion=True)
+        new_schemas = []
+        try:
+            local_machine_name = socket.gethostname().\
+                replace('-','_').replace('.','_')
+            for characters in characters_list:
+                schema_name = u'mssqlcli_{0}_{1}_{2}'.\
+                    format(local_machine_name, random_str(), characters)
+                query = u'CREATE SCHEMA {0}'.format(schema_name)
+                for _, _, _, _, is_error in client.execute_query(query):
```

Comment:
Do we need a for loop?

Reply:
unfortunately yes. client.execute_query(query) is a generator.